### ISSUE TITLE
Can't Change Default Notification Icon In Android?

### ISSUE THREAD

**Issue by anastely** [OneSignal/react-native-onesignal#1077]
**Description:**

I'm trying to change the default notification "bell icon" in the app.
I'm following this [guide](https://documentation.onesignal.com/docs/customize-notification-icons) from Documentation.

**Environment**

```
react-native: 0.63.3
react-native-onesignal: 3.9.2
```

**Steps to Reproduce Issue:**

1. Generate icon from [Android Asset studio](https://romannurik.github.io/AndroidAssetStudio/icons-notification.html#source.type=clipart&source.clipart=ac_unit&source.space.trim=1&source.space.pad=0&name=ic_stat_ac_unit). Named as `ic_stat_onesignal_default`.
1. put all folders into `android/app/src/main/res/`.
1. Send the small_icon via Api and Onesignal Dashboard. `small_icon: 'ic_stat_onesignal_default'`
1. Un-install app and clean build folder then rebuild it, Also generated separate Apk.

Nothing works!
And still got the bell icon as a notification icon :)

**Comment by anastely** [comment]
Hey @rgomezp
any help in this?

**Comment by anastely** [comment]
After added 
AndroidManifest.xml 
```
<meta-data
    android:name="com.google.firebase.messaging.default_notification_icon"
    android:resource="@drawable/ic_stat_onesignal_default" /> // I choose mipmap-xxxhdpi icon size :)
```
 i just was wanted to test if firebase meta-data will work or not and Actually it's work and I got the icon I want!!
have anybody **explanation** for this!

Edit:
Actually if I set it to `com.onesignal.messaging.default_notification_icon` will work too.
and you should use the name `ic_stat_onesignal_default` otherways will not work and will appear the bell default icon.

> Possibly wherever you are referencing the android:name from the manifest file, it will tie back to the correct icon regardless of whether you're actually using Firebase. The name/value pair should be all that's important. As long as you have a meta-data tag in the manifest. [@thomasciv stackoverflow :) ](https://stackoverflow.com/users/14539418/thomasciv)

**Comment by mahmoudalaa97** [comment]
> After added
> AndroidManifest.xml
> 
> ```
> <meta-data
>     android:name="com.google.firebase.messaging.default_notification_icon"
>     android:resource="@drawable/ic_stat_onesignal_default" /> // I choose mipmap-xxxhdpi icon size :)
> ```
> 
> i just was wanted to test if firebase meta-data will work or not and Actually it's work and I got the icon I want!!
> have anybody **explanation** for this!
> 
> Edit:
> Actually if I set it to `com.onesignal.messaging.default_notification_icon` will work too.
> and you should use the name `ic_stat_onesignal_default` otherways will not work and will appear the bell default icon.
> 
> > Possibly wherever you are referencing the android:name from the manifest file, it will tie back to the correct icon regardless of whether you're actually using Firebase. The name/value pair should be all that's important. As long as you have a meta-data tag in the manifest. [@thomasciv stackoverflow :) ](https://stackoverflow.com/users/14539418/thomasciv)

Thank You It's Work Form Me ❤️

**Comment by areavisuale** [comment]
I get this error: Resource 'drawable/ic_stat_onesignal_default' not found in Android Manifest.xml:87.
How do you choose the icon? I have many resolutions in different drawable folders. Could this be the problem?

**Comment by rgomezp** [comment]
Howdy @areavisuale ,
This seems like a different problem altogether.

Go to `android/app/src/main/res/drawable` and check if you have a file with name `ic_stat_onesignal_default` or not. If not, make sure the names match.

Cheers

**Comment by anastely** [comment]
Hey @rgomezp,
I guess should the documentation to be updated, With this workaround!
Maybe Many users faced the same issue and that's will help them!

**Comment by rgomezp** [comment]
@anastely ,
Thanks for the feedback. Can we close this issue then?

**Comment by FayyazAliKhan1** [comment]
A gray box is showing to me instead of the icon.. on android 7 it is fine but on 8 9 10 is a gray box.. followed everything above

**Comment by munibkhalidhere** [comment]
@FayyazAliKhan1 did you able to do it

**Comment by FayyazAliKhan1** [comment]
yeah you just have to create the exact sizes of the icon which OneSignal described..Android asset studio will also creates it if you upload a proper image(don't know what resolution image it accepts ).

**Comment by messycoding** [comment]
i am experiencing the same issue, any solution on this issue?

**Comment by AmarVenturit** [comment]
for anyone who is still facing issues. I added the code below to the androidManifest.xml file inside the Application tag.

```
<meta-data android:name="com.onesignal.messaging.default_notification_icon" android:resource="@mipmap/ic_stat_onesignal_default" />
```

To remove the gray box I used `https://romannurik.github.io/AndroidAssetStudio/icons-notification.html#source.type=clipart&source.clipart=ac_unit&source.space.trim=1&source.space.pad=0&name=ic_stat_ac_unit` to create the png notification icon with `ic_stat_onesignal_default` name.

After this copied of the files from download zip folder to the correspond mipmap folders. All DONE!! just run `yarn android` or `npm run android`
